### ISSUE TITLE
[DataTable] A11Y:  when hideScrollIndicator is true, there is no way to scroll the table via the keyboard nor is the scroll visually available

### ISSUE THREAD

**Issue by ginabak** [Shopify/polaris-react-archive#5268]
## Issue summary

The [Data Table Component](https://polaris.shopify.com/components/lists-and-tables/data-table) has the option to enable **hideScrollIndicator**, which hides the keyboard focusable scroll when the table horizontally collapses to be scrollable. The scroll is only visible via a mouse which is not accessible to those only using a keyboard. 

When the `hideScrollIndicator` prop is true, the table looks like this: 

<img width="476" alt="hide scroll indicator" src="https://user-images.githubusercontent.com/43223543/157980793-a00d4ff3-bd99-454e-b015-e3f5c76f7160.png">

## Expected behavior

The best solution might be (when the hideScrollIndicator is true and the table horizontally collapses to be scrollable) to make the tab index with an attribute of zero (which allows the scroll to be keyboard focusable), create a unique id for that scroll taken from the table header and add an aria labelled by (with the same ID). This allows both the screen reader to read the scroll and allow the scroll to be keyboard focusable. 

There are many places in web where the hideScrollIndicator prop is used (due to weird formatting). 

SEE EXAMPLE: 
<img width="586" alt="WEIRD SPACE ISSUE" src="https://user-images.githubusercontent.com/43223543/157980737-a7316f17-bacc-4ba0-8d8c-f42436955a0c.png">

## Steps to reproduce the problem

1. See the example in [code sandbox ](https://lxswwr.csb.app/)
2. Continually shrink the screen horizontally until the table horizontally collapses to be scrollable. 
3. Notice that one is unable to scroll the table with the keyboard, nor is it visually available.
